modules/admin-bar/module.php

```diff
@@ -0,0 +1,132 @@
+<?php
+namespace Elementor\Modules\AdminBar;
+
+use Elementor\Plugin;
+use Elementor\Core\Base\Document;
+use Elementor\Core\Base\Module as BaseModule;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Module extends BaseModule {
+	/**
+	 * @var Document[]
+	 */
+	private $documents = [];
+
+	/**
+	 * @return bool
+	 */
+	public static function is_active() {
+		return is_admin_bar_showing();
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_name() {
+		return 'admin-bar';
+	}
+
+	/**
+	 * Collect the documents that was rendered in the current page.
+	 *
+	 * @param Document $document
+	 * @param $is_excerpt
+	 */
+	public function add_document_to_admin_bar( Document $document, $is_excerpt ) {
+		if (
+			$is_excerpt ||
+			! $document::get_property( 'show_on_admin_bar' ) ||
+			! $document->is_editable_by_current_user()
+		) {
+			return;
+		}
+
+		$this->documents[ $document->get_main_id() ] = $document;
+	}
+
+	/**
+	 * Scripts for module.
+	 */
+	public function enqueue_scripts() {
+		if ( empty( $this->documents ) ) {
+			return;
+		}
+
+		wp_enqueue_script(
+			'elementor-admin-bar',
+			$this->get_js_assets_url( 'admin-bar' ),
+			[ 'elementor-common' ],
+			ELEMENTOR_VERSION,
+			true
+		);
+
+		wp_localize_script(
+			'elementor-admin-bar',
+			'elementor_admin_bar',
+			$this->get_admin_bar_config()
+		);
+	}
+
+	/**
+	 * Creates the config for 'Edit with elementor' menu item.
+	 *
+	 * @return array
+	 */
+	protected function get_edit_button_config() {
+		$queried_object_id = get_queried_object_id();
+		$href = null;
+
+		if ( is_singular() && isset( $this->documents[ $queried_object_id ] ) ) {
+			$href = $this->documents[ $queried_object_id ]->get_edit_url();
+
+			unset( $this->documents[ $queried_object_id ] );
+		}
+
+		return [
+			'id' => 'elementor_edit_page',
+			'title' => __( 'Edit with Elementor', 'elementor' ),
+			'href' => $href,
+			'children' => array_map( function ( $document ) {
+				return [
+					'id' => "elementor_edit_doc_{$document->get_main_id()}",
+					'title' => $document->get_post()->post_title,
+					'sub_title' => $document::get_title(),
+					'href' => $document->get_edit_url(),
+				];
+			}, $this->documents ),
+		];
+	}
+
+	/**
+	 * Creates admin bar menu items config.
+	 *
+	 * @return array
+	 */
+	public function get_admin_bar_config() {
```

Comment:
Use the built-in `get_init_settings()` method.

Call the settings by `get_settings()` method

Reply:
It's not settings that are used on server side, it's a config for the client side.